### ISSUE TITLE
 Request for better installation instructions for TF on Ampere GPUs!

### ISSUE THREAD

**Issue by wittmarf** [tensorflow/tensorflow#47066]
After 6 weeks of trial and error, I gave up and replaced our brand new 3090 cards with the old 2080 Ti. **I just could not get tensorflow on the new GPUs to work.** 

My colleagues and I have worked with TF from its early days. We have always been early adopters of the latest GPU generation. **There was never a mess like this.** In the worst case, the new TF system didn't utilize all the GPU's new resources, but we never had installation troubles like this.

I tried all installation approaches mentioned in the TF home page (python packages, docker containers, compilation from source). Either it failed completely or - after hours/days on Stack Overflow - ended in disappointing results.

The closest I got to a success came with the upgrade to TF 2.4, but I was getting a million times the warning message 
**"your cuda software stack is old. we fallback to the nvidia driver for some compilation. update your cuda version to get the best performance. the ptxas error was: ptxas fatal : value 'sm_86' is not defined for option 'gpu-name'"**
at each start of a TF script.

This is not inspiring confidence in the coupling of TF and CUDA. **This is plainly unusable.** 

As I understand it, the CUDA toolkit is too old compared to the driver, so the driver is recompiled to match the CUDA version. **But why does this happen again and again?** Once should be enough. We have Ubuntu 20.04 with driver 460 installed. It should be possible to go back to version 450, but we could not achieve that. I don't know if this would help, anyway.

With Docker containers, there seem to be the same problems because they use the driver of the host system. 

Another word about working with containers because they are recommended in the installation instructions: We work with an IDE which offers Docker support only for extra cost. No IDE, no debugger (and other important tools). And like this, Docker is not an option.

**Comment by Saduf2019** [comment]
@wittmarf 
Please refer to [this comment](https://github.com/tensorflow/tensorflow/issues/46933#issuecomment-773975025) and let us know.

**Comment by wittmarf** [comment]
I was aware of this possibility, but I don't think hiding error or warning messages is a solution. 

With "not inspiring confidence in the coupling of TF and CUDA", this is exactly what I meant. It would be a solution if the Nvidia PTX compiler was invoked just once and saved its result.

**Comment by ALIANS-PRODUCTIONS** [comment]
OS Windows 10 Pro : 1909 18363.1379
CUDA TOOLKIT : 11.1
CUDNN: 8.1.0

GPU : MSI RTX 3090 x trio
CPU i7 3930k 32GB ram
using Anaconda
python 3.7.9
tf-nightly-gpu 2.5.0.dev20210114
tf-nightly  2.5.0.dev20210114

=> The above config allows me to run my code LSTM code in tensor flow (extract below):
However clone the environment in Anaconda and upgrade the tf-nightly build to anything more recent, including todays build
of 2.5.0.dev20210228 ... when the code reaches tf.model.fit .. it crashes out with no error reporting at all.

The reason why i tried upgrading the tf-nightly build : currently the model training time is about the same as on my RTX 2080ti and sometimes even slower which thought was down to issues with the tf-nightly builds and Ampere.

Would very appreciate  any suggestions on why the newer tf-nightly builds are making my code crash 

code block:

model = tf.keras.models.Sequential([
    #input shape equals term points in curve
    tf.keras.layers.Bidirectional(tf.keras.layers.LSTM(32,return_sequences=True),input_shape=(n_days,n_cols)),
    tf.keras.layers.Dropout(0.2),
    
    tf.keras.layers.Bidirectional(tf.keras.layers.LSTM(32)),
    tf.keras.layers.Dropout(0.2),

    tf.keras.layers.Dense(64, activation='relu',kernel_initializer=initializer),
    tf.keras.layers.BatchNormalization(),
    tf.keras.layers.Dropout(0.2),
        
    tf.keras.layers.Dense(64, activation='relu',kernel_initializer=initializer, bias_initializer=output_bias),
    tf.keras.layers.BatchNormalization(),
    tf.keras.layers.Dropout(0.2),

    tf.keras.layers.Dense(t_cols, activation='sigmoid', bias_initializer=output_bias)])

#list of keras metrics to compute in model
METRICS = [
      keras.metrics.BinaryAccuracy(name='binary_accuracy'),
      keras.metrics.Precision(name='precision'),
      keras.metrics.Recall(name='recall'),
      keras.metrics.AUC(name='auc'),
]

model.compile(optimizer=Adam(learning_rate=lng_rate),
              loss='binary_crossentropy',
              metrics=METRICS)

history=model.fit(
    X_train, 
    Y_train, 
    epochs=training_epochs, 
    batch_size=training_batch_size, 
    verbose=1,
    class_weight=label_weights[0],
    validation_data=(X_validation,Y_validation))

**Comment by jvishnuvardhan** [comment]
@ALIANS-PRODUCTIONS Looks like you may have different issue when compared to the current issue. Can you please open a new issue with the standalone code. Creating new issue may help others who are facing similar issue like you. Thanks!

**Comment by ALIANS-PRODUCTIONS** [comment]
Hi there, I created a separate ticket with sample code attached in the second comment and extensive detail:
Thank you very much for looking at my case!

tf-nightly-gpu builds from February are crashing on 'tf.keras model.fit()' for Nvidia RTX 3090; January builds performance is slower than 2080ti on : 2Layer Bidirectional LSTM + 3 Dense Layers #47473
